test/unit/lib/entities/context/QuoteContextHandler.test.ts

```diff
@@ -115,6 +124,31 @@ describe('QuoteContextManager', () => {
       expect(requests[2]).toMatchObject(QUOTE_REQUEST_DL_EXACT_OUT);
     });
 
+    it.only('merges simulateFromAddress on classic requests', () => {
+      const context1 = new MockQuoteContext(QUOTE_REQUEST_DL);
+      context1.setDependencies([QUOTE_REQUEST_DL_EXACT_OUT, QUOTE_REQUEST_CLASSIC]);
+      const context2 = new MockQuoteContext(QUOTE_REQUEST_CLASSIC);
+      const simulateFromAddress = '0x1111111111111111111111111111111111111111';
+      context2.setDependencies([
+        Object.assign({}, QUOTE_REQUEST_CLASSIC, {
+          config: {
+            routingType: RoutingType.CLASSIC,
+            protocols: ['v3'],
+            simulateFromAddress,
+          },
+          key: QUOTE_REQUEST_CLASSIC.key,
+        }),
+      ]);
+      const handler = new QuoteContextManager([context1, context2]);
+      const requests = handler.getRequests();
+      expect(requests.length).toEqual(3);
+      expect(requests[0]).toMatchObject(QUOTE_REQUEST_DL);
+      // injects simulateFromAddress into the classic request
+      expect(requests[1]).toMatchObject(QUOTE_REQUEST_CLASSIC);
```

Comment:
might be reading this wrong but wouldnt the base request (QUOTE_REQUEST_CLASSIC) have changed after layering the one with simulateFromAddress onto it?

Reply:
matchObject only checks that everything in A is in B. It doesn't check that everything in B is in A. thats why next line is a check that simulate is in the object